gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/remote/EmbeddedRemoteConnection.java

```diff
@@ -59,7 +62,11 @@ public <E> CompletableFuture<RemoteTraversal<?, E>> submitAsync(final GremlinLan
         // the new submit() in 3.3.x when the deprecation is removed
         final CompletableFuture<RemoteTraversal<?, E>> promise = new CompletableFuture<>();
         try {
-            promise.complete(new EmbeddedRemoteTraversal(JavaTranslator.of(g).translate(gremlinLang)));
+            final GremlinLangScriptEngine scriptEngine = new GremlinLangScriptEngine();
+            final Bindings b = new SimpleBindings();
```

Comment:
better to prefer `scriptEngine.createBindings()` which does the same thing right now but if we ever made lang specific bindings it would spare us some change.